### ISSUE TITLE
[Bug]: .templatesyncignore causing error

### ISSUE THREAD

**Issue by JessicaS11** [AndreasAugustin/actions-template-sync#510]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

The presence of a .templatesyncignore file causes the action to fail with commit changes
```  
::info::handle .templatesyncignore
fatal: empty string is not a valid pathspec. please use . instead if you meant to match all paths
```

Removing the file allows the action to proceed. I've tried with the `.templatesyncignore` file in the top level of the repo as well as within `.github/`

### Expected Behavior

I expected the action to run with or without a .templatesyncignore file.

### Current Behavior

In this [test repo](https://github.com/ICESAT-2HackWeek/jupyterbook-template-tester/tree/main) running the action fails as in [this run log](https://github.com/ICESAT-2HackWeek/jupyterbook-template-tester/actions/runs/8442392936/job/23123694239).

### Steps To Reproduce

Run the action with a `.templatesyncignore` file present in the repo. The contents of the file (i.e. an actual list of ignore paths or an empty file) both cause failure. Only removing the file from the repo allows the action to proceed past ::info::handle .templatesyncignore.

### Additional Information/Context

I'm encountering this issue on both v1 and v2 of the action. The updates contain several merge conflicts, but I wouldn't expect those to be a problem (and the action proceeds with the conflicts present in the case where there is no ignore file).

### Template sync version Version

v2

**Comment by JessicaS11** [comment]
Hello @AndreasAugustin - thanks for the lightning fast response! I've tried with an empty and non-empty `.templatesyncignore` file [here](https://github.com/ICESAT-2HackWeek/jupyterbook-template-tester/tree/main), with the same result.

**Comment by AndreasAugustin** [comment]
> Hello @AndreasAugustin - thanks for the lightning fast response! I've tried with an empty and non-empty `.templatesyncignore` file [here](https://github.com/ICESAT-2HackWeek/jupyterbook-template-tester/tree/main), with the same result.

hi @JessicaS11 sorry for the deletion of my above comment. Wanted to do a check :)

thanks for your answer related to my question if the `.templatesyncignore` is empty.

Also thanks for the bug report and the example. Will try to dig into it ASAP.

**Comment by AndreasAugustin** [comment]
@all-contributors please add @JessicaS11 for bug

**Comment by AndreasAugustin** [comment]
@JessicaS11 am not super sure yet about the root cause. But can you try please (as a temporarly mitigation action) to remove the empty lines and change your `.templatesyncignore` to

```txt
# THIS FILE CANNOT BE SYNCED
# use glob patterns as in .gitignore
# recommend listing non-basics tutorial directories here
# configuration files
cookiecutter.yaml
# environment lock files
conda/*lock.yml
# book files (configuration, usage-specific)
book/_config.yml
team/team.yaml
# event-specific tutorials
```


within the action we use [git reset --pathspec-from-file=<file>](https://git-scm.com/docs/git-reset#Documentation/git-reset.txt---pathspec-from-fileltfilegt)
related to docs *pathspec elements are separated by LF or CR/LF. * 

Need to do further checks to find either a solution or update the docs related to the root cause.

**Comment by JessicaS11** [comment]
> remove the empty lines and change your .templatesyncignore

I can confirm this worked - by removing the blank lines the action ran successfully and opened a PR. Thanks!

Please do let me know if I can help with a PR or review. My team and I are excited to use this action, so thanks for building and maintaining it.

**Comment by AndreasAugustin** [comment]
> > remove the empty lines and change your .templatesyncignore
> 
> I can confirm this worked - by removing the blank lines the action ran successfully and opened a PR. Thanks!
> 
> Please do let me know if I can help with a PR or review. My team and I are excited to use this action, so thanks for building and maintaining it.

Thanks a lot for the feedback! Highly appreciated
😺 

Will try to find a generic solution the next days

**Comment by AndreasAugustin** [comment]
@JessicaS11 there is a fix ready in the pipeline #512 
you can test with `AndreasAugustin/actions-template-sync@fix/templatesyncignore`

```yaml
...

  - name: actions-template-sync
        uses: AndreasAugustin/actions-template-sync@fix/templatesyncignore
        with:
          ...
...
```

**Comment by JessicaS11** [comment]
> @JessicaS11 there is a fix ready in the pipeline https://github.com/AndreasAugustin/actions-template-sync/pull/512
you can test with AndreasAugustin/actions-template-sync@fix/templatesyncignore

Woohoo! Thanks - I just tested and it works great. Really appreciate the quick fix.
